### PR TITLE
[Cocoapods] Fixing Snapshot Tests

### LINKED ISSUE #6609: Snapshot tests aren't running after the Cocoapods 1.6.0 update
Snapshot tests are no longer run with the other tests following https://github.com/material-components/material-components-ios/pull/6580

---
#### Internal data
- Associated internal bug: [b/124382701](http://b/124382701)

### LINKED ISSUE #6700: Snapshot Testing Not Working After Cocoapods 1.6.0 Update
After the Cocoapods 1.6.0 update snapshot tests are not being run in Xcode. There are no snapshot test targets included in the MDCDragons/MDCCatalog scheme.

### PR DESCRIPTION
**Pull request by andrewoverton**
Initially this PR was supposed to consist of a revert of #6619 (which reverted #6611) with some small tweaks to fix whatever was wrong in #6611. It turned into something else! I had a hard time figuring out a way to both run MDCDragons/MDCCatalog with ⌘R _and_ run snapshot tests within either of the two apps' schemes. The only way I was able to manage it was by abandoning our convention for podspecs (only for snapshot tests), which relies heavily on nested subspecs and test_specs. In its place, I wrote a probably not very idiomatic ruby module that aims to allow for all snapshot testing source files to funnel into a single unit test target but still give us some flexibility in how we configure snapshot tests on a per component basis.

Closes #6700.

### CODE REVIEW

**Review by yarneo — APPROVED** [review]

File: MaterialComponentsSnapshotTests.podspec [hunk 1]
```diff
@@ -1,139 +1,88 @@
...
-    component.ios.deployment_target = '8.0'
-    component.test_spec 'tests' do |tests|
-      tests.test_spec 'snapshot' do |snapshot_tests|
-        snapshot_tests.requires_app_host = true
-        snapshot_tests.source_files = "components/#{component.base_name}/tests/snapshot/*.{h,m,swift}"
-        snapshot_tests.dependency "MaterialComponentsSnapshotTests/private/Snapshot"
-      end
-    end
-  end
+module SnapshotPodspecHelper
```
Comment (codeman7) [inline MaterialComponentsSnapshotTests.podspec:1]: This file is absolutely amazing!

File: MaterialComponentsSnapshotTests.podspec [hunk 2]
```diff
@@ -1,139 +1,88 @@
...
+  # launch because XCTest isn't found.
+  s.source_files = ["components/private/Snapshot/src/SourceDummies/*.{h,m}"]
 
-  s.subspec "private" do |private_spec|
-    private_spec.test_spec "Snapshot" do |snapshot|
-      snapshot.ios.deployment_target = '8.0'
-      snapshot.source_files = "components/private/#{snapshot.base_name}/src/*.{h,m,swift}"
-      snapshot.dependency 'iOSSnapshotTestCase', '2.2.0'
-    end
+  s.test_spec "SnapshotTests" do |snapshot_tests|
```
Comment (romoore) [inline MaterialComponentsSnapshotTests.podspec:80]: So they're basically all bundled into one big target? If it works, sounds good!

File: MaterialComponentsSnapshotTests.podspec [hunk 3]
```diff
@@ -1,139 +1,88 @@
...
-        snapshot_tests.source_files = "components/#{component.base_name}/tests/snapshot/*.{h,m,swift}", "components/#{component.base_name}/tests/snapshot/supplemental/*.{h,m,swift}"
-        snapshot_tests.resources = "components/#{component.base_name}/tests/snapshot/resources/*"
-        snapshot_tests.dependency "MaterialComponentsSnapshotTests/private/Snapshot"
-        snapshot_tests.dependency "MDFInternationalization"
-      end
-    end
-  end
+Pod::Spec.new do |s|
+  s.name         = "MaterialComponentsSnapshotTests"
+  s.version      = "77.0.0"
```
Comment (romoore) [inline MaterialComponentsSnapshotTests.podspec]: Should be 78.0.0 now. :-\

File: components/Cards/BUILD [hunk 4]
```diff
@@ -182,21 +182,9 @@ mdc_snapshot_objc_library(
     ],
 )
 
-swift_library(
```
Comment (romoore) [inline components/Cards/BUILD:185]: This looks unrelated - is this just a clean up for empty sources?

File: components/Ripple/BUILD [hunk 5]
```diff
@@ -19,7 +19,10 @@ load(
     "mdc_objc_library",
     "mdc_public_objc_library",
     "mdc_unit_test_suite",
+    "mdc_snapshot_objc_library",
```
Comment (romoore) [inline components/Ripple/BUILD]: Can these BUILD file changes be moved to a separate PR or are they somehow required for the Podspec changes?

File: components/Cards/BUILD (hunk 4 above)
Comment (andrewoverton) [reply components/Cards/BUILD:185]: I got rid of the Cards Swift source dummy file so I had to make this change otherwise CI wouldn't pass

File: components/Ripple/BUILD (hunk 5 above)
Comment (andrewoverton) [reply components/Ripple/BUILD]: In a previous iteration of this PR I had added a Swift source dummy to `components/Ripple/src/` so I had to add the snapshot tests to the BUILD otherwise CI wouldn't pass. I can get rid of this now I guess...
